File: Behavioral - Chain of Responsability/Python/abstract/abstract.py
from abc import abstractmethod
from typing import Any
from interface.interface import Interface


class AbstractInterface(Interface):
  _next_handler: Interface = None

  def set_next(self, handler: Interface) -> Interface:
    self._next_handler = handler
    return handler

  @abstractmethod
  def handle(self, request: Any) -> str:
    if self._next_handler:
      return self._next_handler.handle(request)

    return None

File: Behavioral - Chain of Responsability/Typescript/abstract/abstract.ts
import { Interface } from "../interface/interface.ts";


export abstract class Abstract implements Interface {
  private nextInterface: Interface;

  public setNext(Interface: Interface): Interface {
    this.nextInterface = Interface;
    return Interface;
  }

  public handle(request: string): string | null {
    if (this.nextInterface) {
      return this.nextInterface.handle(request);
    }
    return null;
  }
}